src/org/opensolaris/opengrok/index/IndexDatabase.java

```diff
@@ -1483,6 +1504,44 @@ private void finishWriting() throws IOException {
         }
     }
 
+    private boolean chkFields(File file, String path) throws IOException {
+        int n = 0;
+        postsIter = uidIter.postings(postsIter);
+        while (postsIter.nextDoc() != DocIdSetIterator.NO_MORE_DOCS) {
+            ++n;
+            // Read a limited-fields version of the document.
+            Document doc = reader.document(postsIter.docID(), CHECK_FIELDS);
+            if (doc == null) {
+                LOGGER.log(Level.FINER, "No Document: {0}", path);
+                continue;
+            }
+
+            /**
+             * Verify TABSIZE, or return a value to indicate mismatch.
+             * For an older OpenGrok index that does not yet have TABSIZE,
+             * ignore the check so that no extra work is done. After a re-index,
+             * the TABSIZE check will be active.
+             */
+            int reqTabSize = project != null ? project.hasTabSizeSetting() ?
```

Comment:
Nested ternary operator, it's very bad for readability, maybe rewrite like this?

```java
 project != null && project.hasTabSizeSetting() ? project.getTabSize() : 0;
```

Reply:
fixed